packages/affiliates/examples/dapp-mining/web3-tagging.js

```diff
@@ -0,0 +1,59 @@
+const Web3 = require("web3");
+const { getAbi } = require("@uma/core");
+const argv = require("minimist")(process.argv.slice(), { string: ["emp", "tag", "tokens", "collateral"] });
+
+const web3 = new Web3();
+
+const {
+  // example emp for uusd weth
+  emp = "0x3605Ec11BA7bD208501cbb24cd890bC58D2dbA56",
+  // this should technically be a legit payout address for dapp mining rewards
+  tag = "0xa0dfF1F09a0DCEda7641b7eB07c3BDbe66a06C35",
+  // how many tokens to mint
+  tokens = "100",
+  // how much collateral to back the tokens
+  collateral = "100000"
+} = argv;
+
+// This is an example of creating a position in an EMP, the only function which needs to be tagged with your address
+function createData(empAddress, collateralToSend, tokensToCreate) {
+  const emp = new web3.eth.Contract(getAbi("ExpiringMultiParty"));
+  const encodedData = emp.methods
+    .create({ rawValue: web3.utils.toWei(collateralToSend) }, { rawValue: web3.utils.toWei(tokensToCreate) })
+    .encodeABI();
+  return encodedData;
+}
+
+// This takes encoded data field and just appends the tag
+function tagData(data, tag) {
+  // convert tag to hex and remove the 0x prefix
+  return data.concat(web3.utils.toHex(tag).slice(2));
```

Comment:
Would like to maybe hash the tag so that we can use any kind of string, obscure address and produces fixed length output. What do you guys think of this? Currently this just converts a string to hex.

Reply:
What would be the hash secret that we could use to decode it?

Reply:
i just mean use `web3.utils.keccak256(string)`. Since we have a whitelist of the addresses we can still match on the other end. its kind of nice since then we have a fixed length string and can hash arbitrary data.

Reply:
im going to just leave this as is for now